examples/screenshot.c

```diff
@@ -181,8 +171,6 @@ static void write_image(const char *filename, int width, int height) {
 		free(output);
 	}
 
-	argb_to_rgba(data, height, buffer_stride);
```

Comment:
What you're doing here is against the spec. The `wl_buffer` is created with the format `WL_SHM_FORMAT_XRGB8888`, so we shouldn't put data in another format in there. That also breaks compatibility with other screenshooters (this protocol is shared with orbital).

Reply:
orbital-screenshooter is what I tested this with, and it came out fine.
I came to the conclusion that *we* were the ones doing something wrong.
![test](https://user-images.githubusercontent.com/5235114/31877706-fb97cd42-b833-11e7-9791-b691a1dd06bd.png)

Reply:
We must figure out why. The screenshooter is requesting data in the format `WL_SHM_FORMAT_XRGB8888`, and then using ImageMagick with `bgra`. There's something wrong somewhere.

Reply:
Indeed, it seems that they're not respecting the spec.

* They're shuffling pixel data around: https://github.com/giucam/orbital/blob/master/src/screenshooter/main.cpp#L308
* Solid black is `0x000000FF`, not `0xFF000000`: https://github.com/giucam/orbital/blob/master/src/screenshooter/main.cpp#L280

Reply:
It'll have to do with some endianness-related crap.
I think this comment from weston provides some context: https://github.com/wayland-project/weston/blob/903721a6215f474787b5daf02761fbcb1d3a0bb5/libweston/pixel-formats.h#L56

Reply:
I don't think DRM can be used as an excuse to store `WL_SHM_FORMAT_ARGB8888` in little-endian. We should respect the format of the `wl_buffer` (that is, ARGB in sequential order) regardless of what the backend is, just like clients don't store little-endian data in buffers when they want to render something.

Reply:
The wayland formats are compatible with both the DRM and GBM formats.
WL_SHM_FORMAT_ARGB8888 is defined to be little endian as part of the spec. Go read <wayland-server-protocol.h>

Reply:
My bad, you're right.

Reply:
Do we need to file a bug with orbital?